src/realm/transaction.cpp

```diff
@@ -514,6 +514,35 @@ void Transaction::upgrade_file_format(int target_file_format_version)
             t->migrate_sets_and_dictionaries();
         }
     }
+    if (current_file_format_version < 24) {
+        // rewrite the string indexes with the comparison order
+        for (auto k : table_keys) {
+            auto t = get_table(k);
+            t->migrate_string_sets(); // rewrite sets to use the new string order
```

Comment:
Yes the migration could take some time and it is proportional to the number of strings in sets. Not sure there is a solution to avoid it at this point.